### ISSUE TITLE
lock is released when the task is skipped

### ISSUE THREAD

**Issue by eur00t** [artemistomaras/django-ethereum-events#11]
Hi maintainers,

In the current implementation, if the task is running, the first concurrent task removes the lock, so the next one after that will come through.
Is this the intended behaviour?

I think the lock should prevent any concurrent tasks from executing, so the goal is to only run one task at any given time.

**Comment by artemistomaras** [comment]
Hi,

The intented behaviour is to indeed prevent any concurrent tasks from executing.

If the task is running, the first concurrent task that runs, executes the following statement:
https://github.com/artemistomaras/django-ethereum-events/blob/master/django_ethereum_events/tasks.py#L59
thus it does not remove the lock. (**acquired=False** since **cache.add** fails).

Can you verify this against the output of your celery worker?

**Comment by artemistomaras** [comment]
@eur00t  I think you are indeed right.

```
Received task: app.tasks.interledger_scheduler_new[90d633bf-6f40-4ea9-b0c8-d0f9d9a45f18] (long running task)...
Received task: django_ethereum_events.tasks.event_listener[e8e44261-eee6-4836-b2be-96511c1be0a7] (first concurrent task)
Event listener is already running. Skipping execution. (this is printed by the first concurrent task and the it releases the lock **MISTAKE**)
(now here the second concurrent task will run while the first task is also running and set the lock again)
```

**Comment by eur00t** [comment]
@artemistomaras thanks for swift reply. I can confirm: I also see this in my celery log.

**Comment by artemistomaras** [comment]
I released version **4.1.1**.

Thank you for your contribution.
